301 add year query parameter to title ii program and top level endpoints

ylyangtw opened the pull request:
```
http://localhost:5000/pdl/titles/title-ii/summary?start_year=2014&end_year=2021
```
```
http://localhost:5000/pdl/titles/title-ii/state-distribution?start_year=2014&end_year=2021
```
```
http://localhost:5000/pdl/titles/title-ii/programs/eqip/state-distribution?start_year=2020&end_year=2021
```
```
http://localhost:5000/pdl/titles/title-ii/programs/eqip/summary?start_year=2020&end_year=2021
```
```
http://localhost:5000/pdl/titles/title-ii/programs/eqip/practice-names?start_year=2020&end_year=2021
```
```
http://localhost:5000/pdl/titles/title-ii/programs/crp/summary?start_year=2021&end_year=2021
```
```
http://localhost:5000/pdl/titles/title-ii/programs/crp/state-distribution?start_year=2021&end_year=2021
```
```
http://localhost:5000/pdl/titles/title-ii/programs/acep/state-distribution?start_year=2021&end_year=2021
```
```
http://localhost:5000/pdl/titles/title-ii/programs/acep/summary?start_year=2021&end_year=2021
```
```
http://localhost:5000/pdl/titles/title-ii/programs/rcpp/state-distribution?start_year=2021&end_year=2021
```
```
http://localhost:5000/pdl/titles/title-ii/programs/rcpp/summary?start_year=2021&end_year=2021
```

sandeep-ps commented:
Please resolve the merge conflicts and use the variables in configs.py to define min and max years. Thanks.

ylyangtw commented:
> Please resolve the merge conflicts and use the variables in configs.py to define min and max years. Thanks.

Thanks! I have resolved the conflicts and changed the variables. Also added handling if statute doesn't exist in CSP during shorter period of years, the CSP endpoints should work.

ylyangtw commented on app/controllers/pdl.py:
```diff
@@ -469,12 +473,26 @@ def titles_title_ii_summary_search():
+    if end_year is None:
+        end_year = max_year  # Default to latest available year
+
     endpoint_response = generate_title_ii_total_summary_response(title_id, cfg.TITLE_II_START_YEAR, cfg.TITLE_II_END_YEAR)
```
It's been addressed

sandeep-ps replied:
Thanks, @ylyangtw. No issues. I will review this soon.

sandeep-ps reviewed (approved):
Tested with locally running frontend and Postman. The changes look good to me. The query parameters are correctly filtering out years. 

Whenever we add a feature to let users select specific year from the UI, we will need to test thoroughly for the cases when some practice groupings are missing during specific year durations like 2019-2023.

I will approve this PR. Thank you.

navarroc reviewed (approved):
Tested several endpoints with different query year options. Looks good.
